src/lib.rs

```diff
@@ -274,16 +295,19 @@ pub mod pallet {
 					<StateRegister<T>>::insert(
 						channel_id,
 						RegisteredState {
+							phase: Phase::Register,
 							state: state.clone(),
 							timeout,
-							concluded: false,
 						},
 					);
 					Self::deposit_event(Event::Disputed(channel_id, state));
 					Ok(())
 				}
 				Some(dispute) => {
-					ensure!(!dispute.concluded, Error::<T>::AlreadyConcluded);
+					ensure!(
+						dispute.phase == Phase::Register,
+						Error::<T>::RegisterPhaseOver
```

Comment:
I think `RegisterPhaseOver` should be renamed or even split into more error codes. And maybe `Phase::Register` should also be renamed to `Dispute`. I like `AlreadyConcluded`, since it gives the reason why the operation failed, but `RegisterPhaseOver` just states that currently, no channel can be registered, but gives no further information. I would prefer if there error message contained a reason other than "cannot do this operation right now". TooEarly and TooLate are much more fitting in that regard. Maybe `DisputeAlreadySettled` or something would be more fitting.

Reply:
To be honest, I have trouble giving the error a better name. The other reviewer already asked me to change it, which I did (before it was called `WrongPhase`). The point is that here we need to be in the RegisterPhase, but we are apparently not. `DisputeAlreadySettled` suggests that the dispute is settled, but this doesn't need to be the case since we could also be in phase `Progress`. Do you have another suggestion?

Reply:
OK, I see. In my view, a dispute comprises the several phases, it's not just the first phase.